File: src/leetCodeProblems/MathCalculations/PascalTriangle118.java
package leetCodeProblems.MathCalculations;

/**
 * LeetCode - https://leetcode.com/problems/pascals-triangle/
 */

import java.util.ArrayList;
import java.util.List;

public class PascalTriangle118 {

    public List<List<Integer>> generate(int numRows) {

        List<List<Integer>> output = new ArrayList<List<Integer>>();

        for(int i=1; i <= numRows; i++) {

            List<Integer> temp = new ArrayList<>();
            int c = 1;

            for(int j=0; j < i; j++) {

                if (j == 0) {
                    c = 1;
                }
                else {
                    c = c*(i-j)/j;
                }

                temp.add(c);
            }

            output.add(temp);
        }

        return output;
    }

    public static void main(String[] args) {

        PascalTriangle118 obj = new PascalTriangle118();

        System.out.println(obj.generate(5));
    }
}

File: src/leetCodeProblems/DynamicProgramming/PascalTriangle118.go
package main

/*
- LeetCode - https://leetcode.com/problems/pascals-triangle/
- Time - O(n^2)
- Space - O(n^2)
*/

func generate(numRows int) [][]int {
	out := [][]int{}

	for i := 1; i <= numRows; i++ {
		temp := make([]int, i)
		for j := 1; j <= i; j++ {
			if j == 1 || j == i {
				temp[j-1] = 1
			} else {
				temp[j-1] = out[i-2][j-2] + out[i-2][j-1]
			}
		}
		out = append(out, temp)
	}
	return out
}

// func main() {
// 	//log.Println(minOperations(3))
// 	log.Println(generate(5))
// 	log.Println(generate(1))
// 	//log.Println(minOperations(31))
// }
